webapp/src/ts/services/contact-save.service.ts

```diff
@@ -55,6 +56,10 @@ export class ContactSaveService {
     }
 
     const doc = this.prepare(submitted.doc);
+    if (typeof original === 'undefined' && doc.should_create_user === 'true') {
+      doc.user_for_contact = {};
+      doc.user_for_contact.add = { status: UserCreationStatus.READY, roles: [doc.role] };
```

Comment:
Would be nice to move this code into the `create-user-for-contacts.service` (as a `setAddUser` function or something).